backend/Tests/Api/GraphQL/Tokens/TokenTest.cs

```diff
@@ -0,0 +1,260 @@
+    }
+
+    public async Task DisposeAsync()
+    {
+        await _testHelper.DisposeAsync();
+    }
+    
+    [Fact]
+    public async Task GivenTokenWithEvent_WhenQueryToken_ThenTokenWithEvents()
+    {
+        // Arrange
+        const ulong contractIndex = 1UL;
+        const ulong contractSubindex = 0UL;
+        const string tokenId = "42";
+        var amount = new BigInteger(42);
+        var cisEventDataMint = new CisMintEvent
+        {
+            ContractIndex = contractIndex,
+            ContractSubIndex = contractSubindex,
+            TokenAmount = amount,
+            TokenId = tokenId,
+            ToAddress = new ContractAddress(contractIndex, contractSubindex),
+        };
+        var tokenEvent = new TokenEvent(
+            contractIndex, contractSubindex, tokenId, cisEventDataMint
+        );
+        var token = new Token
+        {
+            ContractIndex = contractIndex,
+            ContractSubIndex = contractSubindex,
+            TokenId = tokenId
+        };
+        await DatabaseFixture.TruncateTables("graphql_tokens", "graphql_account_tokens", "graphql_token_events");
+        await using (var context = _databaseFixture.CreateGraphQlDbContext())
+        {
+            context.TokenEvents.Add(tokenEvent);
+            context.Tokens.Add(token);
+            await context.SaveChangesAsync();
+        }
+        var query = GetTokenQuery(contractIndex, tokenId);
+
+        // Act
+        var result = await _testHelper.ExecuteGraphQlQueryAsync(query);
+
+        // Assert
+        var tokenFromQuery = result["token"].Deserialize<Token>(_jsonSerializerOptions);
+        result["token"]!["tokenEvents"]![0]!["id"] = 0; // Stub the base64 encoded id
+        var tokenEventsFromQuery = result["token"]!["tokenEvents"].Deserialize<List<TokenEvent>>(_jsonSerializerOptions);
+
+        tokenFromQuery.Should().NotBeNull();
+        tokenFromQuery!.TokenId.Should().Be(tokenId);
+        tokenFromQuery!.ContractIndex.Should().Be(contractIndex);
+        tokenFromQuery!.ContractSubIndex.Should().Be(contractSubindex);
+        
+        tokenEventsFromQuery.Should().NotBeNull();
+        tokenEventsFromQuery!.Count.Should().Be(1);
+        var tokenEventFromQuery = tokenEventsFromQuery[0];
+        tokenEventFromQuery.TokenId.Should().Be(tokenId);
+        tokenEventFromQuery.ContractIndex.Should().Be(contractIndex);
+        tokenEventFromQuery.ContractSubIndex.Should().Be(contractSubindex);
+        tokenEventFromQuery.Event.Should().BeOfType<CisMintEvent>();
+        var eventDataMint = (tokenEventFromQuery.Event as CisMintEvent)!;
+        eventDataMint.TokenAmount.Should().Be(amount);
+        eventDataMint.ToAddress.Should().BeOfType<ContractAddress>();
+        var to = (eventDataMint.ToAddress as ContractAddress)!;
+        to.Index.Should().Be(contractIndex);
+        to.SubIndex.Should().Be(contractSubindex);
+    }
+
+    private class AddressConverter : JsonConverter<Address>
```

Comment:
I have done like this, since I want to avoid having a if statement in production which is only there due to test reason. My experience is that it is always easy to add more code but difficult to delete.

Having this converter only in the test class where it is used makes it explicit where and why it is used. Also the parsing done in the test are not parsing we expect in the CCD scan in general.